api/response/schema.py

```diff
@@ -15,40 +14,47 @@ class Meta:
         model = ResponseModel
 
 
+class ResponseInputs(graphene.InputObjectType):
+    question_id = graphene.Int(required=True)
+    rate = graphene.Int()
+    text_area = graphene.String()
+    missing_items = graphene.List(graphene.Int)
+
+
 class CreateResponse(graphene.Mutation):
     class Arguments:
-        question_id = graphene.List(graphene.Int, required=True)
+        responses = graphene.List(
+            ResponseInputs, required=True
+        )
         room_id = graphene.Int(required=True)
-        rate = graphene.Int()
-        text_area = graphene.String()
-        missing_items = graphene.List(graphene.Int)
 
     response = graphene.List(Response)
 
     def mutate(self, info, **kwargs):
         validate_empty_fields(**kwargs)
         query = Room.get_query(info)
+        responses = []
+        errors = []
         room = query.filter_by(id=kwargs['room_id']).first()
         if not room:
             raise GraphQLError("Non-existent room id")
-        question_ids = kwargs.pop('question_id')
-        responses = []
-        questions = Question.get_query(info).filter(
-            QuestionModel.id.in_(question_ids)).all()
-        valid_question_ids = set()
-        for question in questions:
-            valid_question_ids.add(question.id)
+        for each_response in kwargs['responses']:
```

Comment:
Following the clarification by Boniface that you will always have a list of 3 responses, I think we can have a check here to ensure the responses are 3.

Reply:
@caveinn thanks for the feedback, but it may not always be so. We do not control what the Android team sends, if they decide to send one response to one question based on implementation changes on their side, then that response won't be saved if there is a check to ensure it is always 3 responses.